File: 1859.sorting-the-sentence.java
class Solution {
    public String sortSentence(String s) 
    {
        String[] sentenceArray = s.split(" ");
        String[] sortedSentence = new String[sentenceArray.length];

        int index;
        for (String val : sentenceArray)
        {
            index = Character.getNumericValue(val.charAt((val.length()-1)));
           
            sortedSentence[index-1] = val.replace(val.charAt((val.length()-1)), ' ');
        }
        
        String theSentence="";

        for(String val: sortedSentence)
        theSentence += val;

        theSentence = theSentence.trim();
        
        return theSentence;
        
    }
}

File: 1859-sorting-the-sentence/1859-sorting-the-sentence.py
class Solution:
    def sortSentence(self, s: str) -> str:
        sen_list = s.split()
        ord_sen = [""]*len(sen_list)
        for i in sen_list:
            ord_sen[int(i[-1])-1] = i[:-1]
        return " ".join(ord_sen)